tasks/initialise.yml

```diff
@@ -22,27 +23,67 @@
         PGSETUP_INITDB_OPTIONS: >-
           --encoding=UTF8 --locale=en_US.UTF-8 --auth-host=md5
 
-    - name: postgres | postgresql config file
-      template:
+    - name: postgres | check for presence of "Modified by ome postgresql ansible role" in file
+      ansible.builtin.lineinfile:
+        path: "{{ postgresql_dist_confdir }}/postgresql.conf"
+        line: "#Ansible managed"
+        state: absent
+      check_mode: yes
+      changed_when: false
+      register: config_file_changed
+
+    - name: postgres | check that the postgresql.conf.backup file exists
+      ansible.builtin.stat:
+        path: "{{ postgresql_dist_confdir }}/postgresql.conf.org"
+      register: org_file
+
+    - name: postgres | copy a postgresql.conf to postgresql.conf.org
+      become_user: "{{ postgresql_become_user }}"
+      ansible.builtin.copy:
+        src: "{{ postgresql_dist_confdir }}/postgresql.conf"
+        dest: "{{ postgresql_dist_confdir }}/postgresql.conf.org"
+        remote_src: yes
+      when:
+        - not config_file_changed.found
+        - not org_file.stat.exists
+
+    - name: postgres | get the orginal configuration file
+      become_user: "{{ postgresql_become_user }}"
+      import_tasks: org_config_file.yml
+      when:
+        - not org_file.stat.exists
+        - config_file_changed.found
+
+    # read the default postgresql configuration file
+    - name: postgres | get the postgres conf file contents
+      become_user: "{{ postgresql_become_user }}"
+      ansible.builtin.slurp:
+        src: "{{ postgresql_dist_confdir }}/postgresql.conf.org"
+      register: postgres_config_file_contents_o
+
+    - set_fact: postgres_config_file_contents={{ postgres_config_file_contents_o }}
+
+    - name: postgres | copy postgresql config file
+      become_user: "{{ postgresql_become_user }}"
+      ansible.builtin.template:
         dest: >-
           {{ postgresql_dist_confdir }}/postgresql.conf
         src: "{{ postgresql_dist_conf_postgresql_src }}"
         mode: 0644
+        owner: "{{ postgresql_become_user }}"
       notify:
         - restart postgresql
-
-      become_user: "{{ postgresql_become_user }}"
+      when: (not org_file.stat.exists and config_file_changed.found) or  not config_file_changed.found or org_file.stat.exists
```

Comment:
I am having trouble decoding this logic. Is there a scenario where this would evaluate as `false`?

Reply:
I have deleted this redundant condition.